x-pack/plugins/lens/public/app_plugin/csv_download_provider/csv_download_provider.tsx

```diff
@@ -128,28 +167,33 @@ export const downloadCsvShareProvider = ({
         panel: {
           id: 'csvDownloadPanel',
           title: panelTitle,
-          content: (
-            <DownloadPanelContent
-              isDisabled={!csvEnabled}
-              warnings={getWarnings(activeData)}
-              onClick={async () => {
-                await downloadCSVs({
-                  title,
-                  formatFactory: formatFactoryFn(),
-                  activeData,
-                  uiSettings,
-                });
-                onClose?.();
-              }}
-            />
-          ),
+          content: openCsvLensModal,
         },
       },
     ];
   };
 
+  const jobProviderOptions = ({
+    objectType,
+    objectId,
+    isDirty,
+    onClose,
+    shareableUrl,
+    shareableUrlForSavedObject,
+    ...shareOpts
+  }: ShareContext) => {
+    const { sharingData } = shareOpts as unknown as { sharingData: ReportingSharingData };
+
+    const jobProviderOptions: JobParamsProviderOptions = {
+      shareableUrl: isDirty ? shareableUrl : shareableUrlForSavedObject ?? shareableUrl,
+      objectType,
+      sharingData,
+    };
+    return jobProviderOptions;
+  };
   return {
-    id: 'csvDownload',
+    id: 'csv export',
     getShareMenuItems,
+    jobProviderOptions: jobProviderOptions as unknown as JobParamsProviderOptions,
```

Comment:
Same as here - https://github.com/elastic/kibana/pull/171253/files#r1439918059 - let's try to do without this added field in `ShareMenuProvider`.